conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueInterfaceGenerator.java

```diff
@@ -166,7 +166,15 @@ private MethodSpec apiMethod(EndpointDefinition endpointDef, Function<Optional<T
                 .forEach(referenceType -> methodBuilder.addAnnotation(
                         ClassName.get(referenceType.getPackage(), referenceType.getName())));
 
-        endpointDef.getDeprecated().ifPresent(deprecatedDocsValue -> methodBuilder.addAnnotation(Deprecated.class));
+        endpointDef.getDeprecated().ifPresent(_deprecatedValue -> {
+            if (endpointDef.getTags().contains("deprecated-for-removal")) {
```

Comment:
Let's document this tag in the [Endpoint Tags section of the readme](https://github.com/palantir/conjure-java?tab=readme-ov-file#endpoint-tags).

Reply:
ack, done.